homeassistant/components/androidtv_remote/entity.py

```diff
@@ -0,0 +1,91 @@
+"""Base entity for Android TV Remote."""
+from __future__ import annotations
+
+import asyncio
+
+from androidtvremote2 import AndroidTVRemote, ConnectionClosed
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import DOMAIN
+
+
+class AndroidTVRemoteBaseEntity(Entity):
+    """Android TV Remote Base Entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, api: AndroidTVRemote, config_entry: ConfigEntry) -> None:
+        """Initialize the entity."""
+        self._api = api
+        self._host = config_entry.data[CONF_HOST]
+        self._name = config_entry.data[CONF_NAME]
+        self._attr_unique_id = config_entry.unique_id
+        self._attr_is_on = api.is_on
+        device_info = api.device_info
+        assert config_entry.unique_id
+        assert device_info
+        self._attr_device_info = DeviceInfo(
```

Comment:
You can't delete entities still offered by the integration. The standard HA way would be to disable the entity from the frontend.

![image](https://user-images.githubusercontent.com/732514/234299522-f3abd3bd-c4f9-4699-ab90-92356edc3c55.png)

Reply:
@tronikos I had to move callbacks initialization to `async_added_to_hass` since they shouldn't be in the constructor, and shouldn't be called when the entity is disabled. Your original solution may be broken if you disable the remote entity without deleting the whole device.

Reply:
Yes.